cpp/modules/luma.gl/garrow/src/array.h

```diff
@@ -28,32 +28,32 @@
 #include <utility>
 #include <vector>
 
-#include "./util/attribute-descriptor.h"
-#include "luma.gl/webgpu/src/webgpu-helpers.h"
-
 namespace lumagl {
 namespace garrow {
 
 /// \brief Array data structure that manages the backing GPU buffer.
 class Array {
  public:
-  // TODO(ilija@unfolded.ai): In arrow, Array is an abstract class with typed implementaions that get created using
-  // builder API
-  Array(wgpu::Device device, const AttributeDescriptor& descriptor)
-      : _device{device}, _descriptor{std::move(descriptor)} {}
-  Array(wgpu::Device device, const AttributeDescriptor& descriptor, const std::shared_ptr<arrow::Array>& data);
+  // TODO(ilija@unfolded.ai): In arrow, Array is an abstract class with typed implementaions that gets created using
+  // the builder API
+  explicit Array(wgpu::Device device) : _device{device} {}
+  Array(wgpu::Device device, const std::shared_ptr<arrow::Array>& data);
   template <typename T>
-  Array(wgpu::Device device, const AttributeDescriptor& descriptor, const std::vector<T>& data)
-      : Array{device, descriptor} {
+  Array(wgpu::Device device, const std::vector<T>& data) : Array{device} {
```

Comment:
- Right now if I have a raw pointer, I first need to copy it into a `std::vector`?
- Should we allow for raw pointers?

Reply:
The implementation of that would be very straight-forward, however as noted above, I see this as a temporary solution until builder kind of API is in place and this can be aligned with `arrow`. This is the only section marked with `/* Arrow non-compliant API */` thus far